Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -922,6 +922,31 @@ theorem domCongr_toAlgHom (e : G ≃* H) : (domCongr k A e).toAlgHom = mapDomain
 
 end lift
 
+section Linear
+
+variable {A : Type*} [Semiring k] [Semiring A] [Module k A]
+
+variable (k A) in
+/-- Interpret `single i` as a linear map (but with `MonoidAlgebra` in the type, rather than
+`Finsupp`). -/
+def lsingle (i : G) : A →ₗ[k] MonoidAlgebra A G :=
+  Finsupp.lsingle i
+
+lemma lsingle_def (i : G) :
+    lsingle k A i = Finsupp.lsingle i := rfl
+
+@[simp]
+lemma lsingle_apply (i : G) (a : A) :
+    lsingle k A i a = single i a := rfl
+
+@[ext]
+theorem lhom_ext {B : Type*} [AddCommMonoid B] [Module k B]
```

Comment:
The `@[ext]` lemma should be the one about composition with `lsingle`

Reply:
See https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Basic.html#Polynomial.lhom_ext%27 for an example.